### ISSUE TITLE
Player chat does not appear in server logs.

### ISSUE THREAD

**Issue by OddInc** [NebulousCloud/helix#57]
When a player says something in chat, it is not logged in the server console or web console. OOC is logged and LOOC is logged, but not regular chat. Not a big issue, but would be beneficial if fixed.

**Comment by alexgrist** [comment]
This is intentional behaviour, IC chat is very spammy and is excluded from logging. You could definitely re-add it for your own schema/plugin using the PostPlayerSay hook.

**Comment by alexgrist** [comment]
The ```PostPlayerSay``` hook passes ```client, chatType, message, anonymous```
```	
if (isstring(text) and chatType == "ic") then
	ix.log.Add(client, "chat", chatType and chatType:upper() or "??", text)
end
```

Logging IC chat by default is disabled as it's the most used chat type and has no context to messages, it'd make log searching more annoying and hard to follow. Still, easy enough if you need it for yourself.

**Comment by OddInc** [comment]
`function GM:PlayerSay(client, text)
	local chatType, message, anonymous = ix.chat.Parse(client, text, true)

	if (chatType == "ic") then
		if (ix.command.Parse(client, message)) then
			return ""
		end
	end

	if (hook.Run("PrePlayerMessageSend", client, chatType, message, anonymous) == false) then
		return
	end

	text = ix.chat.Send(client, chatType, message, anonymous)

	if (isstring(text) and chatType != "ic") then
		ix.log.Add(client, "chat", chatType and chatType:upper() or "??", text)
	end

	hook.Run("PostPlayerSay", client, chatType, message, anonymous)
	return ""
end`

I tried to add this code to sv_hooks.lua and it didn't seem to work.

**Comment by alexgrist** [comment]
```function Schema:PostPlayerSay(client, chatType, message, anonymous)
	if (isstring(text) and chatType == "ic") then
		ix.log.Add(client, "chat", chatType and chatType:upper() or "??", text)
	end
end```

**Comment by OddInc** [comment]
Thank you so much!

**Comment by OddInc** [comment]
> ```
> function Schema:PostPlayerSay(client, chatType, message, anonymous)
> 	if (isstring(text) and chatType == "ic") then
> 		ix.log.Add(client, "chat", chatType and chatType:upper() or "??", text)
> 	end
> end
> ```

This does not work, and no matter how much I try, IC chat does not appear in console. Sorry to be bothering you with this, as you believe it's not important, but IC chat logs allow us to see if someone is minging in IC chat, or to see if a character did really learn something IC,

**Comment by alexgrist** [comment]
My bad

```
function Schema:PostPlayerSay(client, chatType, message, anonymous)
	if (isstring(message) and chatType == "ic") then
		ix.log.Add(client, "chat", chatType and chatType:upper() or "??", message)
	end
end
```

**Comment by OddInc** [comment]
Thank you. This worked by putting it in the logging.lua plugin.

**Comment by alexgrist** [comment]
Noo, you put it in your schema sv_hooks.lua
